### PR TITLE
Job details page

### LINKED ISSUE #8: Job details page
Add job details page that can be navigated to from the URL or clicking a job in the job feed.

### PR DESCRIPTION
**Pull request by Aaron-Goldman**
Added job details page that displays a list of skills, benefits, and requirements of a job. Job is specified with job id in URL.

Should be merged after #7.

close #8

### CODE REVIEW

File: src/JobDetails/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  if (loading) {
+    return (
+      <CircularProgress />
+    );
+  }
+  if (error) {
+    return (error.message);
+  }
+
+  const j = data.job;
```
Comment (ahmed-gz) [inline src/JobDetails/index.js]: I suggest you use destructing here to get the properties you need directly

File: src/JobDetails/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  return (
+    <Card>
+      <CardHeader title={j.name} subheader={j.company.name} />
+      <CardContent>
+        <Typography variant="body1">
+          {j.description}
+        </Typography>
+        <List>
+          <ListSubheader>Skills</ListSubheader>
+          {j.jobSkills.map((s) => (
```
Comment (ahmed-gz) [inline src/JobDetails/index.js]: You can use destructing here: `{jobSkills.map(({ skill: { id, name } }) => (`

File: src/JobDetails/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            <ListItem key={s.skill.id} dense>
+              <ListItemText>{s.skill.name}</ListItemText>
+            </ListItem>
+          ))}
+          <ListSubheader>Requirements</ListSubheader>
+          {j.jobRequirements.map((r) => (
```
Comment (ahmed-gz) [inline src/JobDetails/index.js]: Destructing same as above

File: src/JobDetails/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            <ListItem key={r.id} dense>
+              <ListItemText>{r.name}</ListItemText>
+            </ListItem>
+          ))}
+          <ListSubheader>Benefits</ListSubheader>
+          {j.jobBenefits.map((b) => (
```
Comment (ahmed-gz) [inline src/JobDetails/index.js]: Destructing same as above

File: src/JobDetails/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  Card, CircularProgress, CardHeader, CardContent,
+  Typography, List, ListItem, ListSubheader, ListItemText,
+} from '@material-ui/core';
+import { QUERY_JOB_DETAILS } from '../graphql/queries';
+
+function JobDetails() {
+  const { id } = useParams();
+
+  const { loading, error, data } = useQuery(QUERY_JOB_DETAILS, {
+    variables: { id: parseInt(id, 10) },
```
Comment (ahmed-gz) [inline src/JobDetails/index.js]: Would this work: `Number(id)` ?

File: src/JobDetails/index.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  Card, CircularProgress, CardHeader, CardContent,
+  Typography, List, ListItem, ListSubheader, ListItemText,
+} from '@material-ui/core';
+import { QUERY_JOB_DETAILS } from '../graphql/queries';
+
+function JobDetails() {
+  const { id } = useParams();
+
+  const { loading, error, data } = useQuery(QUERY_JOB_DETAILS, {
```
Comment (ahmed-gz) [inline src/JobDetails/index.js:13]: `const { loading, error, data: { job } }`

File: src/JobDetails/index.test.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+};
+
+const mockedResponses = [
+  mockedJobResponse,
+];
+
+it('renders the job details page', async () => {
+  render(
+    <MockedProvider mocks={mockedResponses}>
+      <Router initialEntries={[`/job-details/${MOCK_JOB_ID}`]}>
```
Comment (ahmed-gz) [inline src/JobDetails/index.test.js]: For consistency let's use `JOB_DETAILS_BASE_PATH` here

**Review by ahmed-gz — CHANGES_REQUESTED** [review]
Looks good, left some suggestions!

File: src/JobDetails/index.js (hunk 6 above)
Comment (Aaron-Goldman) [reply src/JobDetails/index.js:13]: This one causes a runtime error because `data` is undefined until the query is completed.

Comment (Aaron-Goldman) [reply src/JobDetails/index.js:13]: Using a state variable for `job` and setting up a callback that changes it after the query finishes works.

Comment (ahmed-gz) [reply src/JobDetails/index.js:13]: @Aaron-Goldman I didn't know data can be undefined, so you can ignore my comment, if it works as it it right now I would avoid creating a state variable just for job data, my suggestion was only a code quality improvement and to use destructing as much as possible to use duplication of accessing properties of an object.

**Review by ahmed-gz — APPROVED** [review]
Looks great, feel free to merge it 👍

**Pull request merged**